### PR TITLE
Update Zenodo for Dataset item type

### PR DESCRIPTION
**Pull request by adam3smith**

### CODE REVIEW

File: Zenodo.js [hunk 1]
```diff
@@ -120,36 +127,35 @@ function doWeb(doc, url) {
...
+			if (!ZU.fieldIsValidForType('DOI', item.itemType) && doi) {
 				item.extra = "DOI: " + doi;
 			}
-			//workaround while we don't have proper item type for data
-			//if (schemaType && schemaType.includes("Dataset")) {
-			if (ZU.xpathText(doc, '//span[@class="pull-right"]/span[contains(@class, "label-default") and contains(., "Dataset")]')) {
+
+			// workaround for pre 6.0.24 versions that don't have proper item type for data
+			// if (schemaType && schemaType.includes("Dataset")) {
+			if (!datasetType == "dataset" && ZU.xpathText(doc, '//span[@class="pull-right"]/span[contains(@class, "label-default") and contains(., "Dataset")]')) {
```
Comment (dstillman) [inline Zenodo.js:158]: > !datasetType == "dataset"

This looks like a mistake.

Comment (adam3smith) [reply Zenodo.js:158]: I don't think it is? `datasetType` is [defined here](https://github.com/zotero/translators/pull/3021/files#diff-23ef15323eb9526928ac756f18c84013122c75bdb657077836608330105c9a77R38) and this code should only run where dataset isn't a valid item type, which this tests for.

Comment (dstillman) [reply Zenodo.js:158]: But you're testing whether `!dataset` is equal to a string, which will always be false.

File: Zenodo.js [hunk 2]
```diff
@@ -155,7 +155,7 @@ function scrape(doc, url) {
 
 			// workaround for pre 6.0.24 versions that don't have proper item type for data
 			// if (schemaType && schemaType.includes("Dataset")) {
-			if (!datasetType == "dataset" && ZU.xpathText(doc, '//span[@class="pull-right"]/span[contains(@class, "label-default") and contains(., "Dataset")]')) {
+			if (!(datasetType == "dataset") && ZU.xpathText(doc, '//span[@class="pull-right"]/span[contains(@class, "label-default") and contains(., "Dataset")]')) {
```
Comment (dstillman) [inline Zenodo.js:158]: You're just going for `datasetType != "dataset"`, right? There's no need for parentheses.

Comment (adam3smith) [reply Zenodo.js:158]: Thanks, I'm clearly a bit out of it post workshop ;)

**Comment by dstillman** [comment]
Now it looks like you pasted in another `scrape` function?

**Comment by adam3smith** [comment]
This should now actually be ready. No idea why that was so hard...

**Pull request merged**